### DISCUSSION TITLE
Failed to determine a suitable driver class (Backend)

### DISCUSSION THREAD

**Discussion by senoramarillo** [Project-Books/book-project#990] [Q&A]
Hello,

I followed all step to run the frontend and backend app (IntelliJ Ultimate). 
I successfully run the frontend but I still got an error for the backend.

I also also used -Dspring.profiles.active=dev to run the backend and added few things in application-dev.properties:
**server.port=5000
spring.jpa.hibernate.ddl-auto=validate
spring.datasource.url = jdbc:postgresql://${DB_HOSTNAME}:${DB_PORT}/${DB_NAME}
spring.datasource.username = ${DB_USERNAME}
spring.datasource.password = ${DB_PASSWORD}
spring.datasource.driver-class-name = org.postgresql.Driver
spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.PostgreSQLDialect
spring.flyway.enabled = false**

ERROR MESSAGE:
*****************************
APPLICATION FAILED TO START
***************************
Description:
Failed to configure a DataSource: 'url' attribute is not specified and no embedded datasource could be configured.

Reason: Failed to determine a suitable driver class
Action:
Consider the following:
	If you want an embedded database (H2, HSQL or Derby), please put it on the classpath.
	If you have database settings to be loaded from a particular profile you may need to activate it (the profiles -Dspring.profiles.active=dev are currently **active).

How can I solve it? Any help is appreciated. Thanks

**Comment by senoramarillo** [comment]
Anyone?

**Comment by tt-fan** [comment]
Hi, there. There was a similar ERROR message when I ran the backend app.
Replace ${} with an actual one. 
For example: (It works for me. )
```
spring.datasource.url = jdbc:postgresql://localhost:5432/test
spring.datasource.username = databaseusername
spring.datasource.password = databasepassword
```
Hope this helps.
